internal/node/hybrid/ip_validator.go

```diff
@@ -0,0 +1,221 @@
+		} else if ip.To4() == nil {
+			return nil, fmt.Errorf("invalid IPv6 address %s in --node-ip flag. only IPv4 is supported", ipStr)
+		}
+		return ip, nil
+	}
+
+	//--node-ip flag not set
+	return nil, nil
+}
+
+func ensureNoHostnameOverride(kubeletArgs []string) error {
+	hostnameOverride, err := extractFlagValue(kubeletArgs, hostnameOverrideFlag)
+	if err != nil {
+		return fmt.Errorf("failure while looking for hostname-override: %w", err)
+	}
+	if hostnameOverride != "" {
+		return fmt.Errorf("hostname-override kubelet flag is not supported for hybrid nodes but found override:  %s", hostnameOverride)
+	}
+	return nil
+}
+
+func validateClusterRemoteNetworkConfig(cluster *eks.Cluster) error {
+	if cluster.RemoteNetworkConfig == nil {
+		return fmt.Errorf("remote network config is not set for cluster %s", *cluster.Name)
+	}
+	if cluster.RemoteNetworkConfig.RemoteNodeNetworks == nil {
+		return fmt.Errorf("remote node networks not found in remote network config for cluster %s", *cluster.Name)
+	}
+	return nil
+}
+
+// Validate given node IP belongs to the current host.
+//
+// validateNodeIP adapts the unexported 'validateNodeIP' function from kubelet.
+// Source: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet_node_status.go#L796
+func validateNodeIP(nodeIP net.IP) error {
+	// Honor IP limitations set in setNodeStatus()
+	if nodeIP.To4() == nil && nodeIP.To16() == nil {
+		return fmt.Errorf("nodeIP must be a valid IP address")
+	}
+	if nodeIP.IsLoopback() {
+		return fmt.Errorf("nodeIP can't be loopback address")
+	}
+	if nodeIP.IsMulticast() {
+		return fmt.Errorf("nodeIP can't be a multicast address")
+	}
+	if nodeIP.IsLinkLocalUnicast() {
+		return fmt.Errorf("nodeIP can't be a link-local unicast address")
+	}
+	if nodeIP.IsUnspecified() {
+		return fmt.Errorf("nodeIP can't be an all zeros address")
+	}
+
+	addrs, err := net.InterfaceAddrs()
+	if err != nil {
+		return err
+	}
+	for _, addr := range addrs {
+		var ip net.IP
+		switch v := addr.(type) {
+		case *net.IPNet:
+			ip = v.IP
+		case *net.IPAddr:
+			ip = v.IP
+		}
+		if ip != nil && ip.Equal(nodeIP) {
+			return nil
+		}
+	}
+	return fmt.Errorf("node IP: %q not found in the host's network interfaces", nodeIP.String())
+}
+
+// getNodeIP determines the node's IP address based on kubelet configuration and system information.
+func getNodeIP(kubeletArgs []string, iamNodeName string) (net.IP, error) {
+	// Follows algorithm used by kubelet to assign nodeIP
+	// Implementation adapted for hybrid nodes
+	// 1) Use nodeIP if set (and not "0.0.0.0"/"::")
+	// 2) If the user has specified an IP to HostnameOverride, use it (not allowed for hybrid nodes)
+	// 3) Lookup the IP from node name by DNS
+	// 4) Try to get the IP from the network interface used as default gateway
+	// Source: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/nodestatus/setters.go#L206
+
+	if err := ensureNoHostnameOverride(kubeletArgs); err != nil {
```

Comment:
this is a good idea but i think it's in the wrong place
this method is supposed to give you the ip that kubelet will select for the node
but here we are running a validation specific for eks-hybrid

If so, this probably belong at the Validate or or even ValidateConfig level